internal/integrationtest/compile_1/compile_test.go

```diff
@@ -526,7 +526,7 @@ func compileWithExportBinariesConfig(t *testing.T, env *integrationtest.Environm
 	defer cli.WorkingDir().Join("arduino-cli.yaml").Remove()
 
 	// Test if arduino-cli config file written in the previous run has the `always_export_binaries` flag set.
-	stdout, _, err := cli.Run("config", "dump", "--format", "json")
+	stdout, _, err := cli.Run("config", "dump", "--format", "json", "--config-file", "arduino-cli.yaml")
```

Comment:
All these integration tests are not intended to test the `--config-file` option, it's just boilerplate code to make the test work.

I'd better change `cli.Run(..)` method to transparently add the `--config-file arduino-cli.yaml` flag to the command line (unless specified otherwise) so we don't need to repeat it repeatedly.